src/platforms/macos/macosutils.mm

```diff
@@ -42,11 +42,32 @@
   Q_ASSERT(appId);
 
   NSString* loginItemAppId =
-      QString("%1.login-item").arg(QString::fromNSString(appId)).toNSString();
-  CFStringRef cfs = (__bridge CFStringRef)loginItemAppId;
-
-  Boolean ok = SMLoginItemSetEnabled(cfs, startAtBoot ? YES : NO);
-  logger.debug() << "Result: " << ok;
+    QString("%1.login-item").arg(QString::fromNSString(appId)).toNSString();
+  
+  // For macOS 13 and beyond, register() and unregister() methods
+  // are used for managing login items since SMLoginItemSetEnabled() is deprecated.
+  // For versions prior to macOS 13, SMLoginItemSetEnabled() is used.
+    if (@available(macOS 13, *)) {
```

Comment:
Nit: The alignment of the if and else statments and some of the blocks below need to be changed.